### PR TITLE
feat: Add CDK Example ec2 in vpc with ssm

### PR DESCRIPTION
**Pull request by kristianpaul**
With this pull request I want to  publish a CDK Python example that deploys an EC2 instance taking advantage of AWS services like System Manager, Amazon Linux 2 AMI and  CDK Assets to package scripts that provision instance as needed.

* Own VPC with public subnet (following AWS Defaults for new accounts)
* Based on latest Amazon Linux 2
* System Manager replaces SSH (Remote session available trough the AWS Console or the AWS CLI.)
* Userdata executed from script in S3 bucket (`configuration.sh`).

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: python/ec2/instance/configure.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+#!/bin/sh
```
Comment (rix0rrr) [inline python/ec2/instance/configure.sh:1]: Could do with a comment in this file explaining what it's for.

File: python/ec2/instance/setup.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import setuptools
```
Comment (rix0rrr) [inline python/ec2/instance/setup.py:1]: I guess this setup.py doesn't add much, does it?

File: python/ec2/instance/app.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        asset = Asset(self, "Asset", path=os.path.join(dirname, "configure.sh"))
+        local_path = instance.user_data.add_s3_download_command(
+            bucket=asset.bucket,
+            bucket_key=asset.s3_object_key
+        )
+
+        # Userdata executes script from S3
+        instance.user_data.add_execute_file_command(
+            file_path=local_path,
+            arguments="--verbose -y"
```
Comment (rix0rrr) [inline python/ec2/instance/app.py:56]: ?

File: python/ec2/instance/app.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            )
+
+        # Instance Role and SSM Managed Policy
+        role = iam.Role(self, "InstanceSSM", assumed_by=iam.ServicePrincipal("ec2.amazonaws.com"))
+
+        role.add_managed_policy(iam.ManagedPolicy.from_aws_managed_policy_name("service-role/AmazonEC2RoleforSSM"))
+
+        # Instance
+        instance = ec2.Instance(self, "Instance",
+        instance_type=ec2.InstanceType("t3.nano"),
```
Comment (rix0rrr) [inline python/ec2/instance/app.py:40]: Needs indentation here.

File: python/ec2/instance/app.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+
+class EC2InstanceStack(core.Stack):
+
+    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        # VPC
+        vpc = ec2.Vpc(self, "VPC",
+        nat_gateways=0,
```
Comment (rix0rrr) [inline python/ec2/instance/app.py:21]: Needs indentation here.

File: python/ec2/instance/app.py (hunk 3 above)
Comment (kristianpaul) [reply python/ec2/instance/app.py:56]: good catch, fixing

**Pull request merged**